apps/avatax/src/modules/avatax/order-cancelled/avatax-order-cancelled-payload-transformer.ts

```diff
@@ -1,12 +1,12 @@
 import { z } from "zod";
-import { OrderCancelledPayload } from "../../webhooks/payloads/order-cancelled-payload";
 import { AvataxOrderCancelledTarget } from "./avatax-order-cancelled-adapter";
+import { CancelOrderPayload } from "../../taxes/tax-provider-webhook";
 
 export class AvataxOrderCancelledPayloadTransformer {
   constructor() {}
 
-  transform(avataxId: string, companyCode: string): AvataxOrderCancelledTarget {
-    const transactionCode = z.string().min(1).parse(avataxId);
+  transform(payload: CancelOrderPayload, companyCode: string): AvataxOrderCancelledTarget {
```

Comment:
you use only one field - avataxId - yet you pass entire object. this is antipattern - you need to hack typescript now etc

instead `Pick< CancelOrderPayload, 'avataxId'>` will solve the problem

Reply:
I was thinking about it and thought of putting only `{ avataxId: string }` but `CancelOrderPayload` in this case is only `avataxId: string` so I think it's fine